src/components/dropdown/dropdown.component.ts

```diff
@@ -175,19 +175,31 @@ export default class SlDropdown extends ShoelaceElement {
         return;
       }
 
+      const computeActiveElement = (element: Element | null) => {
+        if (!element?.shadowRoot?.activeElement) return element;
+        return computeActiveElement(element?.shadowRoot?.activeElement);
+      };
+
+      const computeClosestContaining = (element: Element | null, tagName: string) => {
+        const closest = element?.closest(tagName);
+        if (closest !== null) return closest;
+        return computeClosestContaining((element?.getRootNode() as ShadowRoot).host, tagName);
+      };
+
```

Comment:
Thanks for the PR. Would it make sense to use the `activeElements()` utility in `src/internal/active-elements.ts`? At a glance, it seems like this might be doing roughly the same thing.

Reply:
> Thanks for the PR. Would it make sense to use the `activeElements()` utility in `src/internal/active-elements.ts`? At a glance, it seems like this might be doing roughly the same thing.

yes, it makes sense to replace the `computeActiveElement` function with the utility you mentioned. just pushed the changes, thanks for reviewing!